include/sparrow/arrow_array_schema_utils.hpp

```diff
@@ -0,0 +1,186 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <optional>
+#include <ranges>
+#include <tuple>
+#include <type_traits>
+#include <vector>
+
+#include "sparrow/c_interface.hpp"
+#include "sparrow/memory.hpp"
+#include "sparrow/mp_utils.hpp"
+
+#include "mp_utils.hpp"
+
+namespace sparrow
+{
+    template <class T>
+    constexpr int64_t get_size(const T& value)
```

Comment:
Did you als consider cases where `std::size` is the appropriate call to do?

Should we rename to `sparrow::size` to match the standard? (I'm not a fan of get_ here)

Documentation is missing.